tt_metal/hw/firmware/src/erisc.cc

```diff
@@ -72,10 +72,13 @@ void __attribute__((section("erisc_l1_code.1"), noinline)) Application(void) {
             launch_msg_t* launch_msg_address = &(mailboxes->launch[launch_msg_rd_ptr]);
             DeviceValidateProfiler(launch_msg_address->kernel_config.enables);
             DeviceZoneSetCounter(launch_msg_address->kernel_config.host_assigned_id);
+            // TODO: is this redundant since eth gets unicast go...wouldn't be here if not enabled, right?
             enum dispatch_core_processor_masks enables = (enum dispatch_core_processor_masks)launch_msg_address->kernel_config.enables;
             if (enables & DISPATCH_CLASS_MASK_ETH_DM0) {
+                WAYPOINT("R");
                 firmware_config_init(mailboxes, ProgrammableCoreType::ACTIVE_ETH, DISPATCH_CLASS_ETH_DM0);
-                kernel_init();
+                kernel_init(0);
```

Comment:
I don't fully understand why `kernel_init` is getting passed a 0 here, removed from some paths and taking the kernel address in others. Would you mind giving me a quick explanation.

Reply:
erisc is...different, still.  it doesn't init locals/etc like the others.  I could pass it the arg and not use it, figured I'd leave it as 0.